app/controllers/incomeSources/add/AddIncomeSourceStartDateController.scala

```diff
@@ -0,0 +1,249 @@
+                                                   val ec: ExecutionContext)
+  extends ClientConfirmedController with I18nSupport with FeatureSwitching {
+
+  def showUKProperty: Action[AnyContent] = handleRequestMethod(UKProperty)(isGet = true)
+  def showUKPropertyAgent: Action[AnyContent] = handleRequestMethodAgent(UKProperty)(isGet = true)
+  def showForeignProperty: Action[AnyContent] = handleRequestMethod(ForeignProperty)(isGet = true)
+  def showForeignPropertyAgent: Action[AnyContent] = handleRequestMethodAgent(ForeignProperty)(isGet = true)
+  def showSoleTraderBusiness: Action[AnyContent] = handleRequestMethod(SoleTraderBusiness)(isGet = true)
+  def showSoleTraderBusinessAgent: Action[AnyContent] = handleRequestMethodAgent(SoleTraderBusiness)(isGet = true)
+
+  def submitUKProperty: Action[AnyContent] = handleRequestMethod(UKProperty)(isGet = false)
+  def submitUKPropertyAgent: Action[AnyContent] = handleRequestMethodAgent(UKProperty)(isGet = false)
+  def submitForeignProperty: Action[AnyContent] = handleRequestMethod(ForeignProperty)(isGet = false)
+  def submitForeignPropertyAgent: Action[AnyContent] = handleRequestMethodAgent(ForeignProperty)(isGet = false)
+  def submitSoleTraderBusiness: Action[AnyContent] = handleRequestMethod(SoleTraderBusiness)(isGet = false)
+  def submitSoleTraderBusinessAgent: Action[AnyContent] = handleRequestMethodAgent(SoleTraderBusiness)(isGet = false)
+
+  private def handleRequestMethod(incomeSourceType: IncomeSourceType)
+                                 (isGet: Boolean): Action[AnyContent] =
+    (checkSessionTimeout andThen authenticate andThen retrieveNino
+      andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
+      implicit user =>
+        if(isGet) {
+          handleRequest(
+            isAgent = false,
+            incomeSourceType = incomeSourceType
+          )
+        } else {
+          handleSubmitRequest(
+            isAgent = false,
+            incomeSourceType = incomeSourceType
+          )
+        }
+  }
+
+  private def handleRequestMethodAgent(incomeSourceType: IncomeSourceType)
+                                      (isGet: Boolean): Action[AnyContent] = Authenticated.async {
+    implicit request =>
+      implicit user =>
+        getMtdItUserWithIncomeSources(incomeSourceDetailsService) flatMap {
+          implicit mtdItUser =>
+            if(isGet) {
+              handleRequest(
+                isAgent = true,
+                incomeSourceType = incomeSourceType
+              )
+            } else {
+              handleSubmitRequest(
+                isAgent = true,
+                incomeSourceType = incomeSourceType
+              )
+            }
+        }
+  }
+
+  private def handleRequest(isAgent: Boolean, incomeSourceType: IncomeSourceType)
+                   (implicit user: MtdItUser[_]): Future[Result] = {
+
+    (isEnabled(IncomeSources), getCallsAndMessagesKeyPrefix(isAgent, incomeSourceType)) match {
```

Comment:
pls use if/else in this context, as you evaluating getCallsAndMessagesKeyPrefix every time regardless if you need it or not